### PR TITLE
Be resilient to documents/projects with null-filepaths in our persistence layer.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

**Comment by CyrusNajmabadi** [comment]
**Customer scenario**

TypeScript crashes. 

**Bugs this fixes:**

Reported directly from TypeScript.

**Workarounds, if any**

None.

**Risk**

Low.

**Performance impact**

Low.

**Is this a regression from a previous update?**

Yes.

**Root cause analysis:**

Our switch to sqlite introduced a crash for TypeScript.  This is because our persistence layer assumed that all docuemnts/project would have associated non-null filepaths with them.  In some scenarios TypeScript does not supply these values.

Because we've never required that these values be non-null, we need to be resilient to that sort of data flowing through to the persistence layer.

**How was the bug found?**

Direct report from TypeScript.

**Comment by CyrusNajmabadi** [comment]
Tagging @MattGertz

File: src/VisualStudio/CSharp/Test/PersistentStorage/AbstractPersistentStorageTests.cs [hunk 1]
```diff
@@ -386,20 +386,22 @@ private Solution CreateOrOpenSolution()
...
-                solution = solution.AddDocument(DocumentInfo.Create(DocumentId.CreateNewId(project.Id), "Document1", filePath: documentFile));
+                solution = solution.AddDocument(DocumentInfo.Create(DocumentId.CreateNewId(project.Id), "Document1", 
+                    filePath: nullPaths ? null : documentFile));
             }
 
             return solution;
         }
 
-        private IPersistentStorage GetStorage(Solution solution)
+        protected IPersistentStorage GetStorage(Solution solution)
```
Comment (Pilchie) [inline src/VisualStudio/CSharp/Test/PersistentStorage/AbstractPersistentStorageTests.cs]: How come?

Comment (CyrusNajmabadi) [inline src/VisualStudio/CSharp/Test/PersistentStorage/AbstractPersistentStorageTests.cs]: So we can use it from SQLitePersistentStorageTests.cs

**Pull request merged**